.travis/Dockerfile

```diff
@@ -0,0 +1,46 @@
+FROM ubuntu:xenial
+
+RUN apt-get -qq update \
+    && apt-get -qq install -y --no-install-recommends \
+       build-essential=12.* \
+       wget=1.17.* \
+       clang-6.0 \
+       g++-multilib=4:5.3.* \
+       ca-certificates \
+       && update-alternatives --install /usr/bin/clang++ clang++ /usr/bin/clang++-6.0 100 \
+       && update-alternatives --install /usr/bin/clang clang /usr/bin/clang-6.0 100 \
+       && update-alternatives --install /usr/bin/cc cc /usr/bin/clang-6.0 100 \
+       && update-alternatives --install /usr/bin/c++ c++ /usr/bin/clang++-6.0 100 \
+       && update-alternatives --install /usr/bin/cpp cpp /usr/bin/clang++-6.0 100 \
```

Comment:
Since we're setting the compiler for CMake by the full path `/usr/bin/[g++, clang++]`, we probably don't need to run `update-alternatives`